File: src/types/task.ts
export type TaskStatus = 'PENDING' | 'COMPLETED' | 'FAILED';
export type RecoveryAction = 'RETRY' | 'ROLLBACK' | 'IGNORE';

export interface PipelineTask {
  id: string;
  game_id: string;
  task_type: string;
  status: TaskStatus;
  target_id: string | null;
  created_at: string;
  updated_at: string;
}

File: src-tauri/src/domain/task.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, specta::Type)]
#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
pub enum TaskStatus {
    Pending,
    Completed,
    Failed,
}

impl TaskStatus {
    pub fn as_str(&self) -> &'static str {
        match self {
            Self::Pending => "PENDING",
            Self::Completed => "COMPLETED",
            Self::Failed => "FAILED",
        }
    }

    pub fn from_str(s: &str) -> Self {
        match s {
            "COMPLETED" => Self::Completed,
            "FAILED" => Self::Failed,
            _ => Self::Pending,
        }
    }
}

#[derive(Debug, Clone, Serialize, Deserialize, specta::Type)]
pub struct PipelineTask {
    pub id: String,
    pub game_id: String,
    pub task_type: String, // 'apply_collection', 'switch_corridor'
    pub status: TaskStatus,
    pub target_id: Option<String>,
    pub created_at: String,
    pub updated_at: String,
}
